### PR TITLE
feat(keeper): allow git clone for allowed GitHub orgs

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary
- `masc_code_git`에 `clone` 액션 추가. GitHub org 단위 allowlist로 제한.
- `config/tool_policy.toml`의 `[git_clone] allowed_orgs`에서 허용 org 관리 (현재: `jeong-sik`).
- URL 파싱: HTTPS, SSH (`git@`), SSH protocol 3가지 형식 지원.
- clone cwd는 `.worktrees/` 자체 또는 하위 디렉토리 허용 (기존 git 액션보다 완화).
- 타임아웃 120초 (기존 30초 대비 네트워크 지연 고려).

## Security
- 허용되지 않은 org → 거부
- GitHub 외 URL (gitlab, bitbucket 등) → 거부
- `.worktrees/` 외 경로 → 거부
- 플래그 형태(`-`로 시작) 인자 전체 차단 — `--upload-pack`, `--config`, `--separate-git-dir` 등 injection 방지; URL만 전달
- `load_clone_allowed_orgs` 파일 읽기/TOML 파싱 실패 시 `Log.Keeper.warn`으로 경고 로깅 (기존: 묵시적 `[]` 반환)

## Product impact

- Promise affected: `none/internal`
- User-visible change: keeper agent가 허용된 GitHub org의 레포를 `.worktrees/` 내로 clone할 수 있음

## Evidence
- [x] `extract_github_org` 9개 케이스 (HTTPS, SSH, non-github, edge cases) — Alcotest
- [x] `validate_clone_url` 6개 케이스 (allowed/disallowed/empty/non-github/mixed-case) — Alcotest
- [x] security 케이스 5개 (domain spoofing, authority spoofing, percent-encoding, dots, uppercase normalization) — Alcotest
- [x] 빌드 통과
- [x] CI green (0 failed jobs)

## Review evidence

- Copilot code review: 4 threads resolved (git_args injection, silent config load, test comment, test harness)
- CodeQL scan: no issues detected

## Linked issue

### CODE REVIEW

**Comment by jeong-sik** [comment]
@copilot-pull-request-reviewer All 4 review threads resolved with acknowledgment:

1. **git_args injection** — Valid. Will add arg allowlist before merge. Tracked.
2. **Silent config load** — Valid. Will add `Log.warn` on parse failure.
3. **Test comment** — Will update header to match actual coverage.
4. **Test harness** — Will migrate to Alcotest for consistency.

Items 1-2 block merge; 3-4 are cosmetic follow-ups.

File: lib/tool_code_write.ml [hunk 1]
```diff
@@ -282,8 +377,46 @@ let handle_code_git ctx args =
...
+    else if has_flag_args then
+      (false, "clone does not accept flags (security: --upload-pack injection blocked)")
+    else if cwd = "" then
+      (false, "cwd parameter required for clone")
+    else
+      match validate_clone_url ~base_path:ctx.config.Room.base_path url with
+      | Error msg -> (false, msg)
+      | Ok () ->
+        match validate_clone_cwd ctx.config cwd with
+        | Error e -> (false, Types.masc_error_to_string e)
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/tool_code_write.ml]: Already addressed in commit 267d390: only the validated URL is passed to git clone, not git_args. Extra positional args (destination paths) are dropped entirely. Command is now: git clone <quoted-url>. No --separate-git-dir or destination override possible.

File: lib/tool_code_write.ml [hunk 2]
```diff
@@ -63,6 +65,103 @@ let truncate_output s =
...
+        Some org
+
+let validate_clone_url ~base_path url =
+  let allowed = load_clone_allowed_orgs ~base_path in
+  if allowed = [] then
+    Error "No allowed orgs configured for git clone (git_clone.allowed_orgs in tool_policy.toml)"
+  else
+    match extract_github_org url with
+    | None ->
+      Error (Printf.sprintf "Cannot parse GitHub org from URL: %s" url)
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/tool_code_write.ml]: Fixed in 7d5a6bd: error message now reads `"Cannot parse GitHub org from clone URL"` without echoing the URL. The strict prefix matching already rejects URLs with credentials (e.g. `https://user:token@github.com/...` does not start with `https://github.com/`), but defense in depth is the right call here.

File: test/test_tool_code_write_coverage.ml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  (check (option string)) "empty string returns None"
+    None
+    (Tool_code_write.extract_github_org "")
+
+let test_no_repo_path () =
+  (check (option string)) "URL with no path after org"
+    None
+    (Tool_code_write.extract_github_org
+       "https://github.com/jeong-sik")
+
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply test/test_tool_code_write_coverage.ml]: Fixed in 7d5a6bd: added `test_trailing_slash_no_repo` and `test_org_only_dot_git` test cases covering both edge cases.
